docs/architecture/sec-apis.md

```diff
@@ -0,0 +1,93 @@
+
+* A new fingerprint is enrolled
+* All fingerprints are unenrolled
+
+Existing Key objects (secret, private, public, and even certificates) are obtained using the Keystore instance's typical methods (e.g., `.getSecretKey()`).  Note that all of the methods on the Key object that would export the value (e.g., getEncoded()) either throw exception or return `null`; this is true even for PublicKeys.
+
+the "Java Standard" Cipher/Mac/Signature classes are used in Android as they are in any other Java/Kotlin environment.  If the key requires authentication, a `UserNotAuthenticatedException` is thrown; if the key no longer valid (as above), a `KeyPermanentlyInvalidatedException` is thrown.
+
+## Using KeyguardManager
+
+The KeyguardManager is a system service that originally used to lock/unlock the keyboard, but has since expanded to lock/unlock the user's device.  It can only be obtained from a `Context` (e.g., `Activity`).
+
+The most interesting methods here are those that determine if a "strong" security lock (Pattern/PIN/Password) is configured and if the device is currently locked.  Both are boolean values; it is not possible to determine _which_ method of lock is configured.
+
+In addition, the KeyguardManager can be used to prompt the user to enter their Pattern/PIN/Password by way of `createConfirmDeviceCredentialIntent()`; if no "strong" security lock is configured, this method returns `null`.
+
+The Intent is created with optional title and description, then dispatched via `startActivityForResult()` to trigger the device prompts.  Applications receive either `RESULT_OK` (if successfully unlocked) or `RESULT_CANCELED` (device prompt is dismissed) via the overridden method `onActivityResult()`.  It is important to note that dispatching and monitoring is best done from a long-running Activity, such as the MainActivity, or the result is never received.
+
+## Using FingerprintManager
+
+**NOTE:** This API is marked as **deprecated** as of API 28 (Android Pie) and replaced with `BiometricPrompt`.
+
+The FingerprintManager is a system service used to interact with a device's fingerprint hardware.  This was added in API 23, and is now deprecated as of API 28.  As with `KeyguardManager` it can only be obtained from a `Context` (e.g., `Activity`).  It also requires the `USE_FINGERPRINT` or `USE_BIOMETRIC` (added in API 28) permission in the app's manifest.
+
+There are methods to determine if fingerprint authentication is possible; detecting if hardware exists and there is at least one fingerprint enrolled.
+
+Engaging the fingerprint reader is done by calling `authenticate()`.  Before doing so, the app must provide a `FingerprintManager.CryptoObject` and a `FingerprintManager.AuthenticationCallback`.  This method immediately; further interaction happens via the passed-in AuthenticationCallback.  An optional `CancellationSignal` can be provided to disengage the fingerprint hardware (e.g., from the user clicking a "Cancel" button).
```

Comment:
"This method immediately" seems like a word or something stopped short here, or am I misreading it?